tests/CppWinRTTests.cpp

```diff
@@ -142,3 +143,120 @@ TEST_CASE("CppWinRTTests::CppWinRTConsistencyTest", "[cppwinrt]")
     // NOTE: C++/WinRT maps other 'std::exception' derived exceptions to E_FAIL, however we preserve the WIL behavior
     // that such exceptions become HRESULT_FROM_WIN32(ERROR_UNHANDLED_EXCEPTION)
 }
+
+#if (defined(__cpp_lib_coroutine) && (__cpp_lib_coroutine >= 201902L)) || defined(_RESUMABLE_FUNCTIONS_SUPPORTED)
+
+// Define our own custom dispatcher that we can force to be have in certain ways.
+// wil::resume_foreground supports any dispatcher that has a dispatcher_traits.
+
+namespace test
+{
+    enum class TestDispatcherPriority
+    {
+        Normal = 0,
+        Weird = 1,
+    };
+
+    using TestDispatcherHandler = winrt::delegate<>;
+
+    enum class TestDispatcherMode
+    {
+        Dispatch,
+        RaceDispatch,
+        Orphan,
+        Fail,
+    };
+
+    struct TestDispatcher
+    {
+        TestDispatcher() = default;
+        TestDispatcher(TestDispatcher const&) = delete;
+
+        TestDispatcherMode mode = TestDispatcherMode::Dispatch;
+        TestDispatcherPriority expected_priority = TestDispatcherPriority::Normal;
+
+        void TryEnqueue(TestDispatcherPriority priority, TestDispatcherHandler const& handler) const

```

Comment:
https://docs.microsoft.com/en-us/uwp/api/windows.system.dispatcherqueue.tryenqueue?view=winrt-20348

Probably need a mode where `TryEnqueue` says it's displeased by returning 'false'. Docs say that it returns false when the dispatcher has been shut down, so that's presumably under app control (or can be known by the app.)

Reply:
DispatcherQueue might return `true` but still end up never invoking the handler. The return value is useless to us. (Returning `false` means "Your handle is definitely not going to be invoked', but `true` means "Your handler is probably going to be invoked, but maybe not.") No need to add extra logic for the `false` case since it's already covered by the `true` case.